### ISSUE TITLE
issue in web app

### ISSUE THREAD

**Issue by gdem123** [arangodb/arangodb#2316]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [ x] 3.0
- [ ] self-compiled devel branch

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ x] own infrastructure
- [ ] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ x] MacOS, version: sierrra

### this is an AQL-related issue:
[ ] I'm using graph features

I'm issuing AQL via:
- [x ] web interface with this browser:     running on this OS: sierra
- [ ] arangosh
- [ ] this Driver:

I've run `db._explain("<my aql query>")` and it didn't shed more light on this.
The AQL query in question is:

I AM USING YOUR STEP BY STEP GETTING STARTED AND THE WEB APP SHOWS ME SOMETHING WEIRD WHEN I SWITCH FROM JSON VIEW TO TABLE VIEW
AFTER HAVING CREATED 3 USERS:
JSON VIEW:
[
  {
    "_key": "1364",
    "_id": "users/1364",
    "_rev": "_Uj9_7pe---",
    "name": "Katie Foster",
    "age": 27
  },
  {
    "_key": "956",
    "_id": "users/956",
    "_rev": "_Uj88mLG---",
    "age": 48,
    "name": "myname"
  },
  {
    "_key": "1488",
    "_id": "users/1488",
    "_rev": "_Uj9BX1G---",
    "name": "James Hendrix",
    "age": 69
  }
]

TABLE VIEW
key	_id	_rev	name	age
1364	users/1364	_Uj9_7pe---	Katie Foster	27
956	users/956	_Uj88mLG---	48	myname
1488	users/1488	_Uj9BX1G---	James Hendrix	69

AS YOU CAN SEETHE NAME AND AGE ARE INVERTED........

The issue can be reproduced using this dataset:

Please provide a way to create the dataset to run the above query on; either by a gist with an arangodump, or `db.collection.save({my: "values"}) statements. If it can be reproduced with one of the ArangoDB example datasets, it's a plus.

### this is a web interface-related issue:
I'm using the web interface with this browser:     running on this OS:
- [ ] authentication is enabled?
- [ ] using the cluster?
- [ ] _system database?

These are the steps to reproduce:
1) open the browser on http://127.0.0.1:8529
2) log in as ...
3) use database [ ] `_system` [ ] other: 
4) click to ...
...

The following problem occurs: [Screenshot?] 
Instead I would be expecting: 

### this is an installation-related issue:
Describe which steps you carried out, what you expected to happen and what actually happened.

**Comment by jsteemann** [comment]
From the data reported this looks like an issue in 3.1, not 3.0.
I reproduced the issue with the test data provided. The AQL query editor's table view does not map the attribute values to the correct columns in the table. It seems to display attribute values in arbitrary columns.

A potential fix for this is:
```diff
diff --git a/js/apps/system/_admin/aardvark/APP/frontend/js/views/queryView.js b/js/apps/system/_admin/aardvark/APP/frontend/js/views/queryView.js
index fecf4c1..c0fd8c1 100644
--- a/js/apps/system/_admin/aardvark/APP/frontend/js/views/queryView.js
+++ b/js/apps/system/_admin/aardvark/APP/frontend/js/views/queryView.js
@@ -2320,7 +2320,7 @@
 
       var callback = function (error) {
         if (error) {
-          arangoHelper.arangoError('Query', 'Could not reload Queries');
+          arangoHelper.arangoError('Query', 'Could not reload queries');
         } else {
           self.updateLocalQueries();
           self.updateQueryTable();
@@ -2379,21 +2379,31 @@
       };
 
       var first = true;
-      var part = [];
-      // self.tableDescription.rows.push(;
+      var headers = {}; // quick lookup cache
+      var pos = 0;
       _.each(data.original, function (obj) {
         if (first === true) {
           tableDescription.titles = Object.keys(obj);
+          tableDescription.titles.forEach(function (t) {
+            headers[String(t)] = pos++;
+          });
           first = false;
         }
-        _.each(obj, function (val) {
+        var part = Array(pos);
+        _.each(obj, function (val, key) {
+          if (!headers.hasOwnProperty(key)) {
+            // different attribute
+            return;
+          }
+
           if (typeof val === 'object') {
             val = JSON.stringify(val);
           }
-          part.push(val);
+
+          part[headers[key]] = val;
         });
+
         tableDescription.rows.push(part);
-        part = [];
       });
 
       $('#outputTable' + counter).append(this.table.render({content: tableDescription}));
```

**Comment by jsteemann** [comment]
@hkernbach : please review and commit if it's working. 3.1 is also affected.

**Comment by hkernbach** [comment]
Thanks, applied your patch. Will be fixed in next bugfix release and 3.2+.

**Comment by dothebart** [comment]
This bugfix is available with ArangoDB 3.1.16 - which is available for download.
